core/src/erasure.rs

```diff
@@ -41,237 +41,231 @@
 //!
 //!
 use crate::packet::{Blob, SharedBlob, BLOB_HEADER_SIZE};
-use std::cmp;
-use std::convert::AsMut;
-use std::sync::{Arc, RwLock};
+use std::{
+    borrow::BorrowMut,
+    sync::{Arc, RwLock},
+};
 
 use reed_solomon_erasure::ReedSolomon;
 
-//TODO(sakridge) pick these values
-/// Number of data blobs
-pub const NUM_DATA: usize = 8;
-/// Number of coding blobs; also the maximum number that can go missing.
-pub const NUM_CODING: usize = 8;
-/// Total number of blobs in an erasure set; includes data and coding blobs
-pub const ERASURE_SET_SIZE: usize = NUM_DATA + NUM_CODING;
+/// Max number of data blobs in an erasure set; Also max number of parity blobs.
+pub const MAX_SET_SIZE: usize = 255;
 
 type Result<T> = std::result::Result<T, reed_solomon_erasure::Error>;
 
-/// Represents an erasure "session" with a particular configuration and number of data and coding
-/// blobs
-#[derive(Debug, Clone)]
-pub struct Session(ReedSolomon);
-
-/// Generates coding blobs on demand given data blobs
-#[derive(Debug, Clone)]
-pub struct CodingGenerator {
-    /// SharedBlobs that couldn't be used in last call to next()
-    leftover: Vec<SharedBlob>,
-    session: Arc<Session>,
+#[derive(Clone, Copy, Default, Debug, PartialEq, Serialize, Deserialize)]
+pub struct CodingHeader {
```

Comment:
comments for these would help the next poor sucker who picks up erasure ;)

Reply:
will fix